File: phoenix_react_curated_list/web/models/blogs.ex
defmodule PhoenixReactCuratedList.Blogs do
  use PhoenixReactCuratedList.Web, :model
  schema "blogs" do
    field :title, :string
    field :subtitle, :string
    field :image, :string
    field :link, :string
    field :author, :string
    timestamps()
  end

end

File: phoenix_react_curated_list/web/static/js/containers/Blogs.js
import React from "react";
import BlogCard from '../presentationals/BlogCard';
import axios from "axios";

class Blogs extends React.Component {
  constructor() {
    super();
    this.state = { blogs: [] };
  }

  componentWillMount() {
    axios.get('http://localhost:4000/blogs')
      .then(response => {
        this.setState({ blogs: response.data.blogs });
      })
      .catch(error => {
        console.log(error);
      });
  }

  render() {
    const posts = this.state.blogs.map((blog, index) =>
      <BlogCard
        key = { index }
        title = { blog.title }
        subtitle = { blog.subtitle }
        image = { blog.image }
        link = { blog.link }
        author = { blog.author }
      />
    );
    return (
      <div>
      {posts}
      </div>
    )
  }
}

export default Blogs
